Added Azure Detekt doc (#422)
appcircleio/appcircle-docusaurus: docs/workflows/android-specific-workflow-steps/azure-bot-for-detekt-report.md

---
title: Azure DevOps Bot for Detekt Report
description: The Azure DevOps Bot for Detekt Report step analyze your Detekt report and post the report details within the opened pull request in Azure DevOps.
tags: [Azure-DevOps, Detekt-Report, analysis]
sidebar_position: 5
---

import Screenshot from '@site/src/components/Screenshot';

# Azure DevOps Bot for Detekt Report

> The **Azure DevOps Bot for Detekt Report** step allows you to analyse your [Detekt report](https://detekt.dev/docs/introduction/reporting/) and post the report details under the opened pull request in [Azure DevOps](https://learn.microsoft.com/en-us/azure/devops/). Additionally, you can modify the [pull request status](https://learn.microsoft.com/en-us/rest/api/azure/devops/git/pull-request-statuses). Integration requires following some necessary steps.

Reviewer: ```suggestion
The **Azure DevOps Bot for Detekt Report** Report step analyzes your [Detekt report](https://detekt.dev/docs/introduction/reporting/) and posts the details to an open pull request in [Azure DevOps](https://learn.microsoft.com/en-us/azure/devops/). It also allows you to modify the [pull request status](https://learn.microsoft.com/en-us/rest/api/azure/devops/git/pull-request-statuses). This step requires preliminary setup and integration.
```

:::caution

> [Danger](https://danger.systems) works on a similar logic to this step, allowing you to utilise the [Danger step](https://docs.appcircle.io/workflows/common-workflow-steps/#code-reviews-with-danger) for other providers like [GitHub](https://github.com), [GitLab](https://about.gitlab.com), or [Bitbucket](https://bitbucket.org/product/guides/getting-started/overview#a-brief-overview-of-bitbucket). Nonetheless, it's worth noting that Danger does not offer support for Azure DevOps at this time.

Reviewer: ```suggestion
[Danger](https://danger.systems/) operates on a similar principle, allowing use of the [Danger step](https://docs.appcircle.io/workflows/common-workflow-steps/#code-reviews-with-danger) with platforms such as [GitHub](https://github.com/), [GitLab](https://about.gitlab.com/), and [Bitbucket](https://bitbucket.org/product/guides/getting-started/overview#a-brief-overview-of-bitbucket). However, Danger currently does not support Azure DevOps.
```

:::

### Prerequisites

The workflow steps that need to be executed before running the **Azure DevOps Bot for Detekt Report** workflow step, along with their respective reasons, are listed in the table below.

| Prerequisite Workflow Step                       | Description                                      |
 |-------------------------------------------------|--------------------------------------------------|
 | [Detekt](https://docs.appcircle.io/workflows/android-specific-workflow-steps/#detekt) | In order to generate the [Detekt report](https://detekt.dev/docs/introduction/reporting/), the **Detekt** step must be executed beforehand.  |

<Screenshot url='https://cdn.appcircle.io/docs/assets/android-workflow-components-azure-bot-for-detekt-report_1.png'/>

### Input Variables
For each step, specific input variables are required for its operation on your system. The input variables necessary for the **Azure DevOps Bot for Detekt Report** are as follows:

<Screenshot url='https://cdn.appcircle.io/docs/assets/android-workflow-components-azure-bot-for-detekt-report_2.png'/>

:::caution

> Confidential information should be entered as a [**secret environment variable**](https://docs.appcircle.io/environment-variables/managing-variables#adding-key-and-text-based-value-pairs). Also, ensure that the [**environment variable group**](https://docs.appcircle.io/environment-variables/managing-variables#using-environment-variable-groups-in-builds) is selected in the [Configuration](https://docs.appcircle.io/build/build-profile-configuration/).

Reviewer: ```suggestion
Enter confidential information as [**secret environment variable**](https://docs.appcircle.io/environment-variables/managing-variables#adding-key-and-text-based-value-pairs) and ensure the appropriate [**environment variable group**](https://docs.appcircle.io/environment-variables/managing-variables#using-environment-variable-groups-in-builds) is selected in the [Configuration](https://docs.appcircle.io/build/build-profile-configuration/).
```
